File: samples/server/petstore/springboot-useoneofextensibleenum/src/main/java/org/openapitools/configuration/EnumConverterConfiguration.java
package org.openapitools.configuration;

import org.openapitools.model.OtherCountryEnum;
import org.openapitools.model.SubsetCountry;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.convert.converter.Converter;

@Configuration(value = "org.openapitools.configuration.enumConverterConfiguration")
public class EnumConverterConfiguration {

    @Bean(name = "org.openapitools.configuration.EnumConverterConfiguration.otherCountryEnumConverter")
    Converter<String, OtherCountryEnum> otherCountryEnumConverter() {
        return new Converter<String, OtherCountryEnum>() {
            @Override
            public OtherCountryEnum convert(String source) {
                return OtherCountryEnum.fromValue(source);
            }
        };
    }
    @Bean(name = "org.openapitools.configuration.EnumConverterConfiguration.subsetCountryConverter")
    Converter<String, SubsetCountry> subsetCountryConverter() {
        return new Converter<String, SubsetCountry>() {
            @Override
            public SubsetCountry convert(String source) {
                return SubsetCountry.fromValue(source);
            }
        };
    }

}

File: samples/server/petstore/kotlin-springboot-multipart-request-model/src/main/kotlin/org/openapitools/configuration/EnumConverterConfiguration.kt
package org.openapitools.configuration

import org.openapitools.model.MultipartMixedStatus

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.convert.converter.Converter

/**
 * This class provides Spring Converter beans for the enum models in the OpenAPI specification.
 *
 * By default, Spring only converts primitive types to enums using Enum::valueOf, which can prevent
 * correct conversion if the OpenAPI specification is using an `enumPropertyNaming` other than
 * `original` or the specification has an integer enum.
 */
@Configuration(value = "org.openapitools.configuration.enumConverterConfiguration")
class EnumConverterConfiguration {

    @Bean(name = ["org.openapitools.configuration.EnumConverterConfiguration.multipartMixedStatusConverter"])
    fun multipartMixedStatusConverter(): Converter<kotlin.String, MultipartMixedStatus> {
        return object: Converter<kotlin.String, MultipartMixedStatus> {
            override fun convert(source: kotlin.String): MultipartMixedStatus = MultipartMixedStatus.forValue(source)
        }
    }

}
